eng/scripts/test-matrix/test-matrix-functions.ps1

```diff
@@ -0,0 +1,317 @@
+Set-StrictMode -Version "4.0"
+
+function CreateDisplayName([string]$parameter, [hashtable]$displayNames) {
+    $name = $parameter
+
+    if ($displayNames[$parameter]) {
+        $name = $displayNames[$parameter]
+    }
+
+    return $name -replace "[\-\._]", ""
```

Comment:
Do we want to collapse `_`? 

Do we know if this is the full set of unsupported chars in job names? Looking at https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#multi-job-configuration 

>Matrix configuration names (like US_IE above) must contain only basic Latin alphabet letters (A-Z, a-z), numbers, and underscores (_). They must start with a letter. Also, they must be 100 characters or less.

We might what to account for the other rules like max chars and starting with a char. Perhaps we should update the regex to be something like `[^A-Za-z_]", ""`

Reply:
That's better, I'll change the regex. My thinking around excluding underscore was since I was using it as the delimiter between parameters, having underscores in values might make the matrix configuration hard to understand from the display name.